jwt/jwt.go

```diff
@@ -124,7 +127,23 @@ func (js jwtSource) Token() (*oauth2.Token, error) {
 	}
 	h := *defaultHeader
 	h.KeyID = js.conf.PrivateKeyID
-	payload, err := jws.Encode(&h, claimSet, pk)
+	var err error
+	payload := ""
+	if js.conf.SignerProvider == nil {
+		var pk *rsa.PrivateKey
+		pk, err = internal.ParseKey(js.conf.PrivateKey)
+		if err != nil {
+			return nil, err
+		}
+		payload, err = jws.Encode(&h, claimSet, pk)
+	} else {
+		var signer jws.Signer
+		signer, err = js.conf.SignerProvider(h.KeyID)
+		if err != nil {
+			return nil, err
+		}
+		payload, err = jws.EncodeWithSigner(&h, claimSet, signer)
+	}
```

Comment:
I toyed with the idea of this to get around the awkward scoping of `err`, but I don't know it's particularly idiomatic Go... Thoughts 
```suggestion
	var encode func(*jws.Header, *jws.ClaimSet) (string, error)
	if js.conf.SignerProvider == nil {
		encode = func(h *jws.Header, c *jws.ClaimSet) (string, error) {
			pk, err := internal.ParseKey(js.conf.PrivateKey)
			if err != nil {
				return "", err
			}
			return jws.Encode(h, c, pk)
		}
	} else {
		encode = func(h *jws.Header, c *jws.ClaimSet) (string, error) {
			signer, err := js.conf.SignerProvider(h.KeyID)
			if err != nil {
				return "", err
			}
			return jws.EncodeWithSigner(h, c, signer)
		}
	}
	payload, err := encode(&h, claimSet)
```

Reply:
@a-h